app/src/gecko/java/org/mozilla/focus/web/LocalizedContentGecko.java

```diff
@@ -0,0 +1,169 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.focus.web;
+
+import android.content.Context;
+import android.content.pm.PackageInfo;
+import android.content.pm.PackageManager;
+import android.content.res.Resources;
+import android.net.Uri;
+import android.support.annotation.NonNull;
+import android.support.v4.util.ArrayMap;
+
+import org.mozilla.focus.R;
+import org.mozilla.focus.locale.Locales;
+import org.mozilla.focus.utils.HtmlLoader;
+import org.mozilla.focus.utils.SupportUtils;
+import org.mozilla.gecko.GeckoSession;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.Map;
+
+public class LocalizedContentGecko {
```

Comment:
\[PMD\] INFO: The class 'LocalizedContentGecko' has a Cyclomatic Complexity of 4 \(Highest = 5\)\.